src/SprykerEco/Yves/ImageSearchAi/Controller/ImageSearchController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Yves\ImageSearchAi\Controller;
+
+use Generated\Shared\Transfer\ImageSearchFindTermsErrorMessageTransfer;
+use Generated\Shared\Transfer\ImageSearchFindTermsErrorResponseTransfer;
+use Generated\Shared\Transfer\ImageSearchFindTermsResponseTransfer;
+use Spryker\Service\UtilText\Model\Url\Url;
+use Spryker\Yves\Kernel\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Csrf\CsrfToken;
+
+/**
+ * @method \SprykerEco\Yves\ImageSearchAi\ImageSearchAiFactory getFactory()
+ */
+class ImageSearchController extends AbstractController
+{
+    /**
+     * @var string
+     */
+    protected const REQUEST_BODY_CONTENT_KEY_IMAGE = 'image';
+
+    /**
+     * @var string
+     */
+    protected const REQUEST_BODY_CONTENT_KEY_TOKEN = '_token';
+
+    /**
+     * @var string
+     */
+    protected const CSRF_TOKEN_ID = 'image_search_ai_csrf';
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function findTermsAction(Request $request): JsonResponse
+    {
+        $requestBodyContent = $this->getFactory()
+            ->getUtilEncodingService()
+            ->decodeJson((string)$request->getContent(), true);
+
+        if (
+            !isset($requestBodyContent[static::REQUEST_BODY_CONTENT_KEY_TOKEN]) ||
+            !$this->getFactory()->getCsrfTokenManager()->isTokenValid(
+                new CsrfToken(static::CSRF_TOKEN_ID, $requestBodyContent[static::REQUEST_BODY_CONTENT_KEY_TOKEN]),
+            )
+        ) {
+            return $this->createAjaxErrorResponse([
+                'error' => 'form.csrf.error.text',
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        unset($requestBodyContent[static::REQUEST_BODY_CONTENT_KEY_TOKEN]);
+
+        $errors = $this->getFactory()->createBase64ImageValidator()->validate($requestBodyContent);
+        if (count($errors)) {
+            $errorResponse = new ImageSearchFindTermsErrorResponseTransfer();
+            $errorResponse->setError('Bad request');
+            foreach ($errors as $error) {
+                $errorMessage = new ImageSearchFindTermsErrorMessageTransfer();
+                $errorMessage->setMessage($error->getMessage());
+                $errorResponse->addImageSearchFindTermsErrorMessage($errorMessage);
+            }
+
+            return new JsonResponse($errorResponse->toArray(), Response::HTTP_BAD_REQUEST);
+        }
```

Comment:
Combine all these validators into one file, also please make sure that after failing Yves page works correctly (without 404 redirect)